backend/src/apiserver/client/argo_fake.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	argoprojv1alpha1 "github.com/argoproj/argo/pkg/client/clientset/versioned/typed/workflow/v1alpha1"
+)
+
+type FakeArgoClient struct {
+	workflowClientFake FakeWorkflowClientInterface
```

Comment:
Can you use FakeWorkflowClient directly? So you don't need define FakeWorkflowClientInterface.

Reply:
If the FakeWorkflowClient is used here, it is not possible to replace it with the FakeBadWorkflowClient, correct?

Reply:
I think a better approach is to create two Fake struct for ArgoClientInterface, one return good WorkflowClientFake one return bad. So you don't need SetFakeWorkflow() function below.

Reply:
Done.